### PR TITLE
working on fix for adv. DS & algos

### PR DESCRIPTION
**Pull request by bonham000**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Closes #XXXX

#### Description

**Revisions to Advanced Data Structures & Algorithms Section at the end of Beta**

This PR:

- Reorders the adv. DS and algos challenges to match the original Issue
- Moves adv. sorting algos from adv. DS to adv. algos section.
- Fixes problems related to `challengeType` and test formatting.
- Verifies challenge tests work for Adv. sorting algos, typed arrays, maps, sets (partially), queues (except for circular queue), trees, and graphs (except for BFS and DFS).
- **I'm going to create an Issue to track what remains to be done for this section (a good fit of work).**

### CODE REVIEW

**Comment by no-stack-dub-sack** [comment]
There's quite a bit here but from what I can tell...
# LGTM!

File: seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json [hunk 1]
```diff
@@ -448,6 +447,185 @@
...
+        "};"
+      ],
+      "tail": [
+        "function isSorted(arr) {",
+        "  let check = (i) => (i == arr.length - 1) ? true : (arr[i] > arr[i + 1]) ? false : check(i + 1);",
+        "  return check(0);",
+        "};"
+      ],
+      "tests": [
+        "console.assert(typeof bubbleSort == 'function', 'message: bubbleSort is a function');",
```
Comment (raisedadead) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: Most challenges that we have use assert, any reason that we are using `console.assert`? I think if this is being used, which is better in my opinion should we be migrating other challenges to this convention as well?

/cc @freeCodeCamp/moderators

Comment (bonham000) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: Oh no, that's actually left over from when I originally wrote this. I meant to have it updated to `assert` like the rest.

Comment (raisedadead) [inline seed/challenges/08-coding-interview-questions-and-take-home-assignments/coding-interview-algorithm-questions.json]: Ah cool, got ya.. this was the debug while drafting the challenges, in that case lets update them.

Thanks.

**Pull request closed without merging**